packages/block-library/src/post-author/edit.js

```diff
@@ -1,25 +1,294 @@
-import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import apiFetch from '@wordpress/api-fetch';
+import { useRef } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import {
+	Notice,
+	PanelBody,
+	SelectControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockColorsStyleSelector from './block-colors-selector';
+
+const DEFAULT_AVATAR_SIZE = 24;
+
+function PostAuthorDisplay( { props, author, authors } ) {
+	const ref = useRef();
+
+	const { editPost } = useDispatch( 'core/editor' );
+
+	const { fontSize } = props;
+	const {
+		TextColor,
+		BackgroundColor,
+		InspectorControlsColorPanel,
+		ColorPanel,
+	} = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', className: 'background-color' },
+		],
+		{
+			contrastCheckers: [
+				{
+					backgroundColor: true,
+					textColor: true,
+					fontSize: fontSize.size,
+				},
+			],
+			colorDetector: { targetRef: ref },
+			colorPanelProps: {
+				initialOpen: true,
+			},
+		},
+		[ fontSize.size ]
+	);
+
+	const {
+		id,
+		firstName,
+		lastName,
+		name,
+		showAvatar,
+		showDisplayName,
+		showByline,
+		showBio,
+		byline = __( 'Written by:' ),
+	} = props.attributes;
+
+	const hasFirstOrLastNameSet = !! firstName || !! lastName;
+	const avatarSizes = [
+		{ value: 24, label: __( 'Small' ) },
+		{ value: 48, label: __( 'Medium' ) },
+		{ value: 96, label: __( 'Large' ) },
+	];
+
+	const changeAuthor = ( authorId ) => {
+		apiFetch( { path: '/wp/v2/users/' + authorId + '?context=edit' } ).then(
+			( newAuthor ) => {
+				editPost( { author: Number( authorId ) } );
+				props.setAttributes( {
+					id: newAuthor.id,
+					name: newAuthor.name,
+					firstName: newAuthor.firstName,
+					lastName: newAuthor.lastName,
+					avatarSize: props.attributes.avatarSize,
+					avatarUrl:
+						newAuthor.avatar_urls[ props.attributes.avatarSize ],
+					description: newAuthor.description,
+				} );
+			}
+		);
+	};
+
+	const authorName =
+		showDisplayName && hasFirstOrLastNameSet
+			? [ firstName, lastName ].join( ' ' )
+			: name;
+	props.setAttributes( {
+		postAuthorName: authorName,
+	} );
+
+	const blockClassNames = classnames( 'wp-block-post-author', {
+		[ fontSize.class ]: fontSize.class,
+	} );
+	const blockInlineStyles = {
+		fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
```

Comment:
If I understand correctly, we can't set  font sizes in this block. Why are the inline styles necessary then?

Reply:
I fogot to add the controls to the inspector. Added!